### PR TITLE
KEYCLOAK-6618 Update German translations

### PR DESCRIPTION
**Pull request by ferdinandhuebner**
- Add missing translations for OTP authenticator settings and update outdated translations for OTP authenticator
- Fix minor issue for the username property (plural -> singular)
- Add missing translations

### CODE REVIEW

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 1]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 federatedIdentityRemovingLastProviderMessage=Sie k\u00F6nnen den letzten Eintrag nicht entfernen, da Sie kein Passwort haben.
 identityProviderRedirectErrorMessage=Fehler bei der Weiterleitung zum Identity Provider.
 identityProviderRemovedMessage=Identity Provider erfolgreich entfernt.
+identityProviderAlreadyLinkedMessage=Die f\u00F6derierte Identit\u00E4t von {0} ist bereits einem anderen Benutzer zugewiesen.
+staleCodeAccountMessage=Die Seite ist abgelaufen. Bitte versuchen Sie es noch einmal.
```
Comment (pilhuhn) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: "Seite abgelaufen" sounds a bit strange.

**Review by pilhuhn — COMMENTED** [review]
In total sounds good. I did not check with the English original messages to verify the content really makes sense, but at least the translations match the keys

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 2]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
...
+totpStep1=Installieren Sie eine der folgenden Applikationen auf Ihrem Smartphone:
+totpStep2=\u00D6ffnen Sie die Applikation und scannen Sie den Barcode.
+totpStep3=Geben Sie den von der Applikation generierten One-Time-Code ein und klicken Sie auf Speichern.
+
+totpManualStep2=\u00D6ffnen Sie die Applikation und geben Sie den folgenden Schl\u00FCssel ein.
+totpManualStep3=Verwenden Sie die folgenden Konfigurations-Werte, falls die Applikation erlaubt sie zu konfigurieren:
+totpUnableToScan=Sie K\u00F6nnen den Barcode nicht scannen?
+totpScanBarcode=Barcode scannen?
+
+totp.totp=Zeitbasiert (time-based)
```
Comment (danielkocot) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: zeitbasiert

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 3]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
...
+totp.hotp=Z\u00E4hlerbasiert (counter-based)
```
Comment (danielkocot) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: z\u00E4hlerbasiert

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 4]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
...
+totpCounter=Z\u00E4hler
 
 missingUsernameMessage=Bitte geben Sie einen Benutzernamen ein.
 missingFirstNameMessage=Bitte geben Sie einen Vornamen ein.
-missingEmailMessage=Bitte geben Sie eine E-Mail Adresse ein.
+invalidEmailMessage=Ung\u00FCltige E-Mail Adresse.
 missingLastNameMessage=Bitte geben Sie einen Nachnamen ein.
+missingEmailMessage=Bitte geben Sie eine E-Mail Adresse ein.
 missingPasswordMessage=Bitte geben Sie ein Passwort ein.
 notMatchPasswordMessage=Passw\u00F6rter sind nicht identisch.
```
Comment (danielkocot) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Die Passw\u00F6rter sind nicht identisch.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 5]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
 federatedIdentityRemovingLastProviderMessage=Sie k\u00F6nnen den letzten Eintrag nicht entfernen, da Sie kein Passwort haben.
 identityProviderRedirectErrorMessage=Fehler bei der Weiterleitung zum Identity Provider.
 identityProviderRemovedMessage=Identity Provider erfolgreich entfernt.
+identityProviderAlreadyLinkedMessage=Die f\u00F6derierte Identit\u00E4t von {0} ist bereits einem anderen Benutzer zugewiesen.
+staleCodeAccountMessage=Die Seite ist abgelaufen. Bitte versuchen Sie es noch einmal.
+consentDenied=Einverst\u00E4ndnis verweigert.
 
 accountDisabledMessage=Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.
```
Comment (danielkocot) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ihr Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 6]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
 
 accountTemporarilyDisabledMessage=Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
```
Comment (danielkocot) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ihr Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.

**Review by danielkocot — CHANGES_REQUESTED** [review]
Corrected translations in the comments.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties (hunk 1 above)
Comment (thomasdarimont) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Vielleicht "Diese Seite ist nicht mehr gültig, bitte versuchen Sie es noch einmal"

Comment (thomasdarimont) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Vielleicht "Diese Seite ist nicht mehr gültig, bitte versuchen Sie es noch einmal"

**Comment by ferdinandhuebner** [comment]
Thanks for the feedback. 
Would you prefer that I use a separate commit to address your comments and rebase and squash after the review is done, or squash right away?

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties (hunk 1 above)
Comment (ferdinandhuebner) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: I like Thomas' suggestion and am going to pick that.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties (hunk 2 above)
Comment (ferdinandhuebner) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Apart from the capitalization, would you suggest that I omit the "(time-based)" part? 

I was uncertain whether to include it or not. If we only display the German word, users might get confused, especially if their authenticator application is not or not properly translated.

Comment (danielkocot) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: You shouldn't omit the "(time-based)" part.

**Comment by ferdinandhuebner** [comment]
I revised the problematic translations. 

If the changes are OK for you, I can rebase and squash so this PR can be merged.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 7]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
 
 configureAuthenticators=Authenticatoren konfigurieren
 mobile=Mobile
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Mobil

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 8]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
...
-totpStep2=\u00D6ffnen Sie die Applikation und scannen Sie den Barcode oder geben Sie den Code ein.
-totpStep3=Geben Sie den von der Applikation generierten One-time Code ein und klicken Sie auf Speichern.
+totpStep1=Installieren Sie eine der folgenden Applikationen auf Ihrem Smartphone:
+totpStep2=\u00D6ffnen Sie die Applikation und scannen Sie den Barcode.
+totpStep3=Geben Sie den von der Applikation generierten One-Time-Code ein und klicken Sie auf Speichern.
+
+totpManualStep2=\u00D6ffnen Sie die Applikation und geben Sie den folgenden Schl\u00FCssel ein.
+totpManualStep3=Verwenden Sie die folgenden Konfigurationswerte, falls Sie diese f\u00FCr die Applikation anpassen k\u00F6nnen:
+totpUnableToScan=Sie k\u00F6nnen den Barcode nicht scannen?
+totpScanBarcode=Barcode scannen?
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Is it a question?

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 9]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
...
 usernameExistsMessage=Der Benutzername existiert bereits.
 emailExistsMessage=Die E-Mail-Adresse existiert bereits.
 
-readOnlyUserMessage=Sie k\u00F6nnen dieses Benutzerkonto nicht \u00E4ndern, da es schreibgesch\u00FCtzt ist.
-readOnlyPasswordMessage=Sie k\u00F6nnen dieses Passwort nicht \u00E4ndern, da es schreibgesch\u00FCtzt ist.
+readOnlyUserMessage=Sie k\u00F6nnen Ihr Benutzerkonto nicht \u00E4ndern, da es schreibgesch\u00FCtzt ist.
+readOnlyUsernameMessage=Sie k\u00F6nnen Ihren Benutzernamen nicht \u00E4ndern, da er schreibgesch\u00FCtzt ist.
+readOnlyPasswordMessage=Sie k\u00F6nnen Ihr Passwort nicht \u00E4ndern, da es schreibgesch\u00FCtzt ist.
 
 successTotpMessage=Mobile Authentifizierung eingerichtet.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Mehrfachauthentifizierung erfolgreich konfiguriert.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 10]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
...
 successTotpRemovedMessage=Mobile Authentifizierung entfernt.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Mehrfachauthentifizierung erfolgreich entfernt.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 11]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
+consentDenied=Einverst\u00E4ndnis verweigert.
 
-accountDisabledMessage=Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.
+accountDisabledMessage=Ihr Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.
 
-accountTemporarilyDisabledMessage=Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
+accountTemporarilyDisabledMessage=Ihr Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
 invalidPasswordMinLengthMessage=Ung\u00FCltiges Passwort\: Minimall\u00E4nge {0}.
-invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
 invalidPasswordMinLowerCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Kleinbuchstaben beinhalten.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort\: Es muss mindestens {0} Kleinbuchstaben beinhalten.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 12]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
+identityProviderAlreadyLinkedMessage=Die f\u00F6derierte Identit\u00E4t von {0} ist bereits einem anderen Benutzer zugewiesen.
+staleCodeAccountMessage=Diese Seite ist nicht mehr gültig, bitte versuchen Sie es noch einmal.
+consentDenied=Einverst\u00E4ndnis verweigert.
 
-accountDisabledMessage=Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.
+accountDisabledMessage=Ihr Benutzerkonto ist gesperrt, bitte kontaktieren Sie den Admin.
 
-accountTemporarilyDisabledMessage=Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
+accountTemporarilyDisabledMessage=Ihr Benutzerkonto ist tempor\u00E4r gesperrt, bitte kontaktieren Sie den Admin oder versuchen Sie es sp\u00E4ter noch einmal.
 invalidPasswordMinLengthMessage=Ung\u00FCltiges Passwort\: Minimall\u00E4nge {0}.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort\: Es muss mindestens {0} Zeichen lang sein.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 13]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
-invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
 invalidPasswordMinLowerCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Kleinbuchstaben beinhalten.
+invalidPasswordMinDigitsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Zahl(en) beinhalten.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort\: Es muss mindestens {0} Zahl(en) beinhalten.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 14]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
 invalidPasswordMinUpperCaseCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Grossbuchstaben beinhalten.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort\: Es muss mindestens {0} Gro\u00DFbuchstaben beinhalten.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 15]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
 invalidPasswordMinSpecialCharsMessage=Ung\u00FCltiges Passwort\: muss mindestens {0} Spezialzeichen beinhalten.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort\: Es muss mindestens {0} Sonderzeichen beinhalten.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 16]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
-invalidPasswordNotUsernameMessage=Ung\u00FCltiges Passwort\: darf nicht gleich sein wie Benutzername.
+invalidPasswordNotUsernameMessage=Ung\u00FCltiges Passwort\: darf nicht gleich sein wie der Benutzername.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort\: Es darf nicht gleich sein wie der Benutzername.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 17]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
 invalidPasswordRegexPatternMessage=Ung\u00FCltiges Passwort\: nicht Regex-Muster (n) entsprechen.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort\: Es entspricht nicht dem Regex-Muster.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 18]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
-invalidPasswordHistoryMessage=Ung\u00FCltiges Passwort: darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.
\ No newline at end of file
+invalidPasswordHistoryMessage=Ung\u00FCltiges Passwort: darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort: Es darf nicht einem der letzten {0} Passw\u00F6rter entsprechen.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 19]
```diff
@@ -133,15 +150,20 @@ federatedIdentityLinkNotActiveMessage=Diese Identit\u00E4t ist nicht mehr aktiv.
...
+invalidPasswordBlacklistedMessage=Ung\u00FCltiges Passwort: Das Passwort ist in der Schwarzen Liste enthalten.
```
Comment (muehlburger) [inline themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Ung\u00FCltiges Passwort: Das Passwort steht auf der Blocklist (schwarzen Liste).

**Review by muehlburger — CHANGES_REQUESTED** [review]
Hi, I reviewed the changes and suggest some minor updates.

**Review by muehlburger — APPROVED** [review]

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 20]
```diff
@@ -24,7 +24,7 @@ familyName=Nachname
 password=Passwort
 passwordConfirm=Passwortbest\u00E4tigung
```
[the remark this answers is not in the document]
Comment (ferdinandhuebner) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: That label is used when the password is changed. The english value of the label is "Confirmation".

I'm going to change it to "Passwort best\u00E4tigen". Other sites, e.g. Google, use that phrase and it is used in the login theme as well.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties [hunk 21]
```diff
@@ -94,29 +97,43 @@ revoke=Berechtigung widerrufen
 
 configureAuthenticators=Authenticatoren konfigurieren
```
[the remark this answers is not in the document]
Comment (ferdinandhuebner) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: I'm happy to change that. However, I would suggest rewording "Authentifikatoren" and "Authenticatoren" to "Mehrfachauthentifizierung" throughout the entire theme. 

It would be strange if we call it "Mehrfachauthentifizierung" when displaying the result of activating or deactivating it, but call it "Authentifikatoren" in the menu and page title. 

TOTP labels are used in the login theme as well. However, it is called "Mobile Authentifizierung" in that theme. 
For the sake of consistency, the same phrase should be used in both themes. Would you prefer "Mobile Authentifizierung" or "Mehrfachauthentifizierung"?

I'll go through the translations of the other themes, because they seem incomplete. Would you prefer that I open another pull-request for hat or update this one?

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties (hunk 20 above)
[the remark this answers is not in the document]
Comment (muehlburger) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: That sounds good.

File: themes/src/main/resources-community/theme/base/account/messages/messages_de.properties (hunk 21 above)
[the remark this answers is not in the document]
Comment (muehlburger) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Definitely "Mehrfachauthentifizierung". It refers to to multi-factor-authentication as well as two-factor authentication.

What does "mobile authentication" mean exactly? I understand it as the possibility to configure two-factor authentication, which uses for example Google Authenticator or FreeOTP. If this was the case it should not be called "Mobile Authentication" but "Mehrfachauthentifizierung".

[the remark this answers is not in the document]
Comment (ferdinandhuebner) [reply themes/src/main/resources-community/theme/base/account/messages/messages_de.properties]: Yes, the term is always used in that context. I reworded it to "Mehrfachauthentifizierung" troughout the account theme and the login theme.

**Comment by ferdinandhuebner** [comment]
As discussed with @muehlburger, I reworded the German translation for multi-factor authentication to "Mehrfachauthentifizierung" instead of "Mobile Authentifizierung".

Commit 0a4059804713b54974c2fae94706d07a35578258 applies the changes we discussed for the account theme to the login theme which was incomplete.

**Review by muehlburger — APPROVED** [review]

**Review by stianst — APPROVED** [review]
